xicam/Acquire/devices/fastccd.py

```diff
@@ -336,33 +337,29 @@ def __init__(self, *args, **kwargs):
         self._readout = 0.080
         super().__init__(*args, **kwargs)
 
-    def set(self, exp):
+    def set(self, exp: float, triggerrate: float = 1., trigger_on: bool = 1, delaytime: float=1e-3, shuttertime: float=140e-3):
         # Exposure time = 0
         # Cycle time = 1
 
         if exp[0] is not None:
             Efccd = exp[0] + self._Tc + self._To
             # To = start of FastCCD Exposure
-            aa = 0  # Shutter open
-            bb = Efccd - self._Tc + aa  # Shutter close
-            cc = self._To * 3  # diag6 gate start
-            dd = exp[0] - (self._Tc * 2)  # diag6 gate stop
-            ee = 0  # Channel Adv Start
-            ff = 0.001  # Channel Adv Stop
-            gg = self._To  # MCS Count Gate Start
-            hh = exp[0] + self._To  # MCS Count Gate Stop
+            tr = triggerrate  # trigger rate
+            tm = trigger_on  # Shutter open
+            dt = delaytime  # delay time
+            st = shuttertime# shutter time
+
 
             # Set delay generator
-            self.parent.dg1.A.set(aa)
-            self.parent.dg1.B.set(bb)
-            self.parent.dg1.C.set(cc)
-            self.parent.dg1.D.set(dd)
-            self.parent.dg1.E.set(ee)
-            self.parent.dg1.F.set(ff)
-            self.parent.dg1.G.set(gg)
-            self.parent.dg1.H.set(hh)
-            self.parent.dg2.A.set(0)
-            self.parent.dg2.B.set(0.0005)
+            self.parent.dg1.trigger_rate.set(tr)
+            self.parent.dg1.trigger_on_off.set(tm)
+            self.parent.dg1.delay_time.set(dt)
+            self.parent.dg1.shutter_time.set(st)
+            self.parent.dg1.init.set()
+            self.parent.dg1.reset.set()
```

Comment:
Is it right to initialize and then immediately reset? What would this mean for the delay generator? Also, can you `.set` without args?

Reply:
ah true, that is useless. Did not think it through.